app/src/main/java/org/dhis2/usescases/searchTrackEntity/SearchRepositoryImplKt.kt

```diff
@@ -44,30 +44,27 @@ class SearchRepositoryImplKt(
         searchParametersModel: SearchParametersModel,
         isOnline: Boolean,
     ): TrackedEntitySearchCollectionRepository {
-        var allowCache = false
-
-        if (searchParametersModel != searchRepositoryJava.savedSearchParameters || !FilterManager.getInstance()
-                .sameFilters(searchRepositoryJava.savedFilters)
-        ) {
-            trackedEntityInstanceQuery =
-                searchRepositoryJava.getFilteredRepository(searchParametersModel)
-        } else {
+        trackedEntityInstanceQuery =
             searchRepositoryJava.getFilteredRepository(searchParametersModel)
-            allowCache = true
-        }
 
-        if (searchRepositoryJava.fetchedTeiUIDs.isNotEmpty() && searchParametersModel.selectedProgram == null) {
-            trackedEntityInstanceQuery =
-                trackedEntityInstanceQuery.excludeUids().`in`(searchRepositoryJava.fetchedTeiUIDs.toList())
+        val allowCache = !(
+            searchParametersModel != searchRepositoryJava.savedSearchParameters ||
+                !FilterManager.getInstance().sameFilters(searchRepositoryJava.savedFilters)
+            )
+
+        if (
+            searchRepositoryJava.fetchedTeiUIDs.isNotEmpty() &&
+            searchParametersModel.selectedProgram == null
+        ) {
+            trackedEntityInstanceQuery = trackedEntityInstanceQuery.excludeUids()
+                .`in`(searchRepositoryJava.fetchedTeiUIDs.toList())
         }
 
-        val pagerFlow = if (isOnline && FilterManager.getInstance().stateFilters.isEmpty()) {
+        return if (isOnline && FilterManager.getInstance().stateFilters.isEmpty()) {
```

Comment:
Can you do a  couple of tests to assert that filter with working list and not filter when no working list selected?